shared/modules/consensus_module.go

```diff
@@ -91,10 +88,11 @@ type ConsensusDebugModule interface {
 
 	SetUtilityUnitOfWork(UtilityUnitOfWork)
 
-	// SetAggregatedStateSyncMetadata is used to set peer's aggregated metadata in testing scenarios to simulate periodic metadata synchronization. It is not intended to be used outside of testing.
-	SetAggregatedStateSyncMetadata(minHeight, maxHeight uint64, peerAddress string)
-	GetAggregatedStateSyncMetadataMaxHeight() (minHeight uint64)
+	// PushStateSyncMetadataResponse is used to simulate metadata received metadata in testing scenarios. It is not intended to be used outside of testing.
+	PushStateSyncMetadataResponse(minHeight, maxHeight uint64)
 
 	// REFACTOR: This should accept typesCons.HotstuffStep and return typesCons.NodeId.
 	GetLeaderForView(height, round uint64, step uint8) (leaderId uint64)
+
+	GetValidatorsAtHeight(height uint64) ([]*types.Actor, error)
```

Comment:
This is not a good design decision. If you look at the implementation of `GetValidatorsAtHeight`, this is all we're doing:

```go
func (m *consensusModule) GetValidatorsAtHeight(height uint64) ([]*coreTypes.Actor, error) {
	readCtx, err := m.GetBus().GetPersistenceModule().NewReadContext(int64(height))
	if err != nil {
		return nil, err
	}
	defer readCtx.Release()
	return readCtx.GetAllValidators(int64(height))
}
```

In short, it's a proxy into the persistence module.

Ask yourself:
1. Who is responsible for knowing the vaalidators at a specific height?
2. Are the validators a part of the state?
3. Is the state a part of persistence?
4. Is the consensus module an orchestator of the state a a view into the state?

In short, you should be able to remove this completely and have a local helper if you need it.

If you need to mock it, mock the persistence module. That's the point of dependency injection.